src/main/java/com/vackosar/gitflowincrementalbuild/boundary/MavenLifecycleParticipant.java

```diff
@@ -68,6 +76,22 @@ public void afterProjectsRead(MavenSession session) throws MavenExecutionExcepti
         logger.info("gitflow-incremental-builder exiting...");
     }
 
+    private boolean isEnabledForBranch(MavenSession session) throws IOException {
+        if (configProvider.get().enabledBranchRegex == null) {
+            return true;
+        }
+        
+        try (GitFactory gitFactory = createGitFactory(session, configProvider.get())) {
+            String branchName = gitFactory.getBranchName();
+            return configProvider.get().enabledBranchRegex.test(branchName);
+        }
+    }
+
+    // only for testing!
+    GitFactory createGitFactory(MavenSession mavenSession, Configuration configuration) throws IOException {
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/gitflow-incremental-builder/gitflow-incremental-builder/pullRequest?prid=5732004)